python/python/lancedb/remote/db.py

```diff
@@ -110,6 +110,9 @@ def open_table(self, name: str) -> Table:
 
         self._client.mount_retry_adapter_for_table(name)
 
+        if index_cache_size is not None:
+            raise ValueError("index_cache_size is not respected by the remote client")
```

Comment:
nit: Maybe just a warning?

Reply:
Good point.  We need to make it as easy as possible to swap between the two and I could see an error here being very annoying.  I changed to a logging message (not even a warning, just `info` since this is technically harmless).